src/game/client/shadow/w3dshadow.h

```diff
@@ -110,6 +110,29 @@ class Shadow
         }
     }
 
+    void Set_Color(int color)
+    {
+        m_opacity = color & 0xFFFFFF;
+
+        if ((m_type & SHADOW_ALPHA_DECAL) != 0) {
+            m_color2 = (color << 24) | m_color1;
+        } else if ((m_type & SHADOW_ADDITIVE_DECAL) != 0) {
+            float o = m_opacity / 255.0f;
+            m_color2 = GameMath::Fast_To_Int_Truncate(((m_color1 >> 16) & 0xFF) * o)
+                | GameMath::Fast_To_Int_Truncate(((m_color1 >> 8) & 0xFF) * o)
+                | GameMath::Fast_To_Int_Truncate((m_color1 & 0xFF) * o);

```

Comment:
I have seen this pattern of color values somewhere else in Thyme. Is it possible that there could be a macro made for this?

Reply:
On second look, are you sure this logic is correct? It looks as if all color channels are pushed into the blue channel. Why?

Reply:
dunno if the logics are correct but pseudo matches

Reply:
Like so would make more sense if I am not mistaken.

Basically what happens is, color value is shifted to lowest 8 bits, masked and scaled by opacity, and then moved back to the byte position it belongs for R G B channel respectively.

```c++
            m_color2 = (GameMath::Fast_To_Int_Truncate(((m_color1 >> 16) & 0xFF) * o) << 16)
                | (GameMath::Fast_To_Int_Truncate(((m_color1 >> 8) & 0xFF) * o) << 8)
                | GameMath::Fast_To_Int_Truncate((m_color1 & 0xFF) * o);
```